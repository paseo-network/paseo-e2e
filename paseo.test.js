import { test, expect } from 'vitest'
import { setupNetworks, withExpect, setupContext, testingPairs } from '@acala-network/chopsticks-testing'
import { ChopsticksProvider, setStorage, setup } from '@acala-network/chopsticks-core'
import { IdbDatabase } from '@acala-network/chopsticks-db/browser'
import { ApiPromise, WsProvider } from "@polkadot/api";

const { check, checkEvents, checkHrmp, checkSystemEvents, checkUmp } = withExpect((x) => ({
  toMatchSnapshot(msg) {
    expect(x).toMatchSnapshot(msg)
  },
  toMatch(value, _msg) {
    expect(x).toMatch(value)
  },
  toMatchObject(value, _msg) {
    expect(x).toMatchObject(value)
  },
}))


test('test', async () => {
   const { alice } = testingPairs()
   const endpoint = 'wss://paseo.rpc.amforc.com'
   const network = await setupContext({timeout: 600000, blockNumber:100,  db: "./db.sqlite", port: 0, endpoint  })

   let version = await network.api.rpc.state.getRuntimeVersion()
   console.log(version.toJSON().specVersion)
   let sudo_key =  await network.api.query.sudo.key();
   await network.dev.setStorage({
    Sudo: {
      key: alice.address
    },
    System: {
      account: [[[alice.address], { providers: 1, data: { free: 1000 * 1e12 } }]],
    },
   })
   await network.api.tx.sudo.sudo(network.api.tx.balances.forceSetBalance(alice.address, 1000000000000)).signAndSend(alice, (result) => console.log(result.txHash.toHuman()));
   await network.dev.newBlock()
 
   await checkSystemEvents(network, 'sudo')
   .redact({ address: true, number: true })
   .toMatchSnapshot()
}, 300000)