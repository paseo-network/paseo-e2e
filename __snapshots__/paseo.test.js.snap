// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test > sudoCall 1`] = `
[
  {
    "data": {
      "sudoResult": "Ok",
    },
    "method": "Sudid",
    "section": "sudo",
  },
]
`;

exports[`test > system events 1`] = `
[
  {
    "data": {
      "sudoResult": "Ok",
    },
    "method": "Sudid",
    "section": "sudo",
  },
]
`;

exports[`test > ttttt 1`] = `
[
  {
    "data": {
      "sudoResult": "Ok",
    },
    "method": "Sudid",
    "section": "sudo",
  },
]
`;
